cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -1605,6 +1622,35 @@ impl Node<Option<cst::Name>> {
     }
 }
 
+/// If this [`ast::Name`] is a known extension function/method name or not
+pub(crate) fn is_known_extension_func_name(name: &ast::Name) -> bool {
+    EXTENSION_STYLES.functions.contains(name)
+        || (name.0.path.is_empty() && EXTENSION_STYLES.methods.contains(&name.basename()))
+}
+
+/// If this [`ast::Id`] is a known extension _method_ name or not
+fn is_known_extension_method_id(id: ast::Id) -> bool {
+    match ast::UnreservedId::try_from(id) {
+        Ok(id) => EXTENSION_STYLES.methods.contains(&id),
+        Err(_) => false, // hardcodes the assumption that no reserved ids are extension method names, but this was already implicit in that `EXTENSION_STYLES.methods` only holds `UnreservedId`s
+    }
+}
+
+/// If this [`ast::Id`] is a known extension function/method name or not
+pub(crate) fn is_known_extension_func_id(id: &ast::Id) -> bool {
+    EXTENSION_STYLES.unqualified_functions.contains(id) || is_known_extension_method_id(id.clone())
+}
+
+/// If this [`SmolStr`] is a known extension function/method name or not
+pub(crate) fn is_known_extension_func_str(s: SmolStr) -> bool {
```

Comment:
I don't particularly like this function since it's not clear if it will treat `s` as an `Id` or a `Name`. It might be better to force callers to do the conversion to `ast::Id`, or maybe it's enough to make it explicit in the name

Reply:
I considered forcing the caller to do the conversion, but the use of `ast::Id::unchecked` is justified based on internals in this file, so I thought it might be better to put the use in this file as well.  However, I'm happy to move it back to the caller, with an appropriate comment.

Reply:
Actually a better idea: let's just keep a hashset of the string versions of all extension function/method names in `EXTENSION_STYLES`

Reply:
looks good, thanks